components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,346 @@
+                        Are you trying to open a Git repository from a self-managed git hoster?{" "}
+                        <a className="text-blue" href={gitpodHostUrl.asAccessControl().toString()}>
+                            Add integration
+                        </a>
+                    </p>
+                </div>
+            );
+        case ErrorCodes.INVALID_GITPOD_YML:
+            return (
+                <div className="mt-2 flex flex-col space-y-8">
+                    <button
+                        className=""
+                        onClick={() => {
+                            createWorkspace({ forceDefaultConfig: true });
+                        }}
+                    >
+                        Continue with default configuration
+                    </button>
+                </div>
+            );
+        case ErrorCodes.NOT_AUTHENTICATED:
+            return (
+                <div className="mt-2 flex flex-col space-y-8">
+                    <button
+                        className=""
+                        onClick={() => {
+                            tryAuthorize(error?.data.host, error?.data.scopes).then((payload) =>
+                                setSelectAccountError(payload),
+                            );
+                        }}
+                    >
+                        Authorize with {error.data.host}
+                    </button>
+                </div>
+            );
+        case ErrorCodes.PERMISSION_DENIED:
+            return <p className="text-base text-gitpod-red w-96">Access is not allowed</p>;
+        case ErrorCodes.USER_BLOCKED:
+            window.location.href = "/blocked";
+            return <></>;
+        case ErrorCodes.NOT_FOUND:
+            return <p className="text-base text-gitpod-red w-96">{error.message}</p>;
+        case ErrorCodes.TOO_MANY_RUNNING_WORKSPACES:
+            return <LimitReachedParallelWorkspacesModal />;
+        case ErrorCodes.NOT_ENOUGH_CREDIT:
+            return <LimitReachedOutOfHours />;
+        case ErrorCodes.INVALID_COST_CENTER:
+            return (
+                <div>
+                    <p className="text-base text-gitpod-red w-96">
+                        The organization <b>{error.data}</b> is not valid.
+                    </p>
+                </div>
+            );
+        case ErrorCodes.PAYMENT_SPENDING_LIMIT_REACHED:
+            return <UsageLimitReachedModal hints={error?.data} />;
+        case ErrorCodes.PROJECT_REQUIRED:
+            return (
+                <p className="text-base text-gitpod-red w-96">
+                    <a className="gp-link" href="https://www.gitpod.io/docs/configure/projects">
+                        Learn more about projects
+                    </a>
+                </p>
+            );
+        default:
+            return <p className="text-base text-gitpod-red w-96">Unknown Error: {JSON.stringify(error, null, 2)}</p>;
+    }
+}
+
+function ExistingWorkspaceModal({
```

Comment:
Ditto on the comment above about using `FunctionComponent<Props>` for defining the component. In this case we're missing the return type, so there are some linter errors that wouldn't get caught.